src/super_gradients/training/models/segmentation_models/stdc/stdc.py

```diff
@@ -394,11 +225,15 @@ class STDCSegmentationBase(SgModule):
     :param dropout: segmentation heads dropout.
     """
 
-    @resolve_param("backbone", BaseFactory({"STDCBackbone": STDCBackbone}))
+    @resolve_param("backbone", BaseFactory(SEGMENTATION_BACKBONES))
+    @resolve_param("initial_upsample", TransformsFactory())
+    @resolve_param("final_upsample", TransformsFactory())
     def __init__(
         self,
-        backbone: AbstractSTDCBackbone,
+        backbone: AbstractSegmentationBackbone,
         num_classes: int,
+        initial_upsample: Optional[nn.Upsample],
```

Comment:
upsample can be done by many ways other than `nn.Upsample` such as, ConvTranspose, PixelShuffle...
I don't think we should force the user to pass those modules as part of the constructor, since some of them requires in_channels as argument, thus are defined dynamically.

Reply:
Tried to mimic pre and post scales.
I believe that we can support other upsample models, as long as we can determine (statically) the parameters. I'll think how to align the signatures

Reply:
I've loosen up the type-hint, [38d5b64](https://github.com/Deci-AI/super-gradients/pull/704/commits/38d5b64d9fa6bd64cd58e5c0f3af079492ab09df).
So basically any `nn.Module` can fit.

If we want to take it to the next level, we can have some sort of `ReshapeFactory` and register `Upsample`, `ConvTranspose`, `PixelShuffle`, etc... WDYT? 
@lkdci 
@BloodAxe ?

Reply:
@NatanBagrov I like your suggestion.

Reply:
Should it be part of this PR?